actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -693,20 +719,43 @@ def generate(route_key, options, recall = {})
                           :original_script_name]
 
       def optimize_routes_generation?
-        default_url_options.empty?
+        !default_url_options
       end
 
       def find_script_name(options)
-        options.delete(:script_name) || ''
```

Comment:
The implementation of this method changed at master. You need to get the updated version.